modules/swagger-codegen/src/main/resources/cpprest/api-source.mustache

```diff
@@ -53,7 +53,22 @@ pplx::task<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/r
     utility::string_t responseHttpContentType;
     
     // use JSON if possible
-    if ( responseHttpContentTypes.size() == 0 || responseHttpContentTypes.find(U("application/json")) != responseHttpContentTypes.end() )
+    if ( responseHttpContentTypes.size() == 0 )
+    {
+        {{#vendorExtensions.x-codegen-response.isPrimitiveType}}
+        responseHttpContentType = U("application/json");
+        {{/vendorExtensions.x-codegen-response.isPrimitiveType}}
+        {{^vendorExtensions.x-codegen-response.isPrimitiveType}}
+        {{#vendorExtensions.x-codegen-response.isString}}
```

Comment:
It looks like the isString would not be similar to isBinary. It is closer to isResponseBinary under 
https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenOperation.java#L17